app/mm-redux/reducers/entities/apps.ts

```diff
@@ -14,11 +14,102 @@ export function bindings(state: AppBinding[] = [], action: GenericAction): AppBi
         validateBindings(action.data);
         return action.data || [];
     }
+    case AppsTypes.CLEAR_APP_BINDINGS:
+        if (state.length > 0) {
+            return [];
+        }
+        return state;
+    default:
+        return state;
+    }
+}
+
+export function bindingsForms(state: AppCommandFormMap = {}, action: GenericAction): AppCommandFormMap {
+    switch (action.type) {
+    case AppsTypes.RECEIVED_APP_BINDINGS:
+        if (Object.keys(state).length) {
+            return {};
+        }
+        return state;
+    case AppsTypes.RECEIVED_APP_COMMAND_FORM: {
+        const {form, location} = action.data;
+        const newState = {
+            ...state,
+            [location]: form,
+        };
+        return newState;
+    }
+    case AppsTypes.CLEAR_APP_BINDINGS: {
+        if (Object.keys(state).length) {
+            return {};
+        }
+        return state;
+    }
+    default:
+        return state;
+    }
+}
+
+export function threadBindings(state: AppBinding[] = [], action: GenericAction): AppBinding[] {
+    switch (action.type) {
+    case AppsTypes.RECEIVED_THREAD_APP_BINDINGS: {
+        validateBindings(action.data.bindings);
```

Comment:
Can we have `validateBindings` return the bindings in order to show that something about the bindings may have been changed by the function?

Reply:
This is suffering from the bug solved by https://github.com/mattermost/mattermost-mobile/pull/5567

Will wait for that to be merged to make this change.